sklearn/metrics/pairwise.py

```diff
@@ -54,6 +55,94 @@ def _return_float_dtype(X, Y):
     return X, Y, dtype
 
 
+def flexible_concatenate(it, final_len=None):
+    """Concatenate the elements of an iterable
+    Supports generators of arrays, lists, sparse matrices or tuples thereof
```

Comment:
PEP257: blank line before this.

Also, should rename "generators" to "iterables"